packages/beacon-node/src/network/reqresp/ReqRespBeaconNode.ts

```diff
@@ -290,13 +323,15 @@ export class ReqRespBeaconNode extends ReqResp {
   }
 
   private async *onStatus(req: ReqRespRequest, peerId: PeerId): AsyncIterable<ResponseOutgoing> {
-    const body = ssz.phase0.Status.deserialize(req.data);
+    const fork = ForkName[ForkSeq[this.currentRegisteredFork] as ForkName];
```

Comment:
needs to be based on the protocol version, not the currently registered fork
use `responseSszTypeByMethod`

Reply:
updated in #8144